recipes/cfcTest.tsx

```diff
@@ -0,0 +1,95 @@
+import { h } from "@commontools/html";
+import {
+  derive,
+  handler,
+  JSONSchema,
+  NAME,
+  recipe,
+  str,
+  UI,
+} from "@commontools/builder";
+
+const UpdaterSchema = {
+  type: "object",
+  properties: {
+    newValues: {
+      type: "array",
+      items: {
+        type: "string",
+        $ifc: { classification: ["confidential"] },
+      },
+    },
+  },
+  title: "Update Values",
+  description: "Append `newValues` to the list.",
+  example: { newValues: ["foo", "bar"] },
+  default: { newValues: [] },
+} as const satisfies JSONSchema;
+
+const InputSchema = {
+  type: "object",
+  properties: {
+    values: {
+      type: "array",
+      items: { type: "string", $ifc: { classification: ["confidential"] } },
+      asCell: true,
+    },
+  },
+  default: {
+    values: ["string"],
+    $ifc: { classification: ["unclassified"], integrity: ["commontools"] },
+  },
+} as const satisfies JSONSchema;
+
+const OutputSchema = {
+  type: "object",
+  properties: {
+    values: { type: "array", items: { type: "string" } },
+    updater: {
+      asStream: true,
+      ...UpdaterSchema,
+    },
+  },
+} as const satisfies JSONSchema;
+
+const updater = handler(
+  UpdaterSchema,
+  InputSchema,
+  (event, state) => {
+    console.log("updating values", event);
+    event.newValues.forEach((value) => {
+      console.log("adding value", value);
+      state.values.push(value);
+    });
+  },
+);
+
+const adder = handler({}, InputSchema, (_, state) => {
+  console.log("adding a value");
+  state.values.push(Math.random().toString(36).substring(2, 15));
+});
+
+export default recipe(InputSchema, OutputSchema, ({ values }) => {
+  derive(values, (values) => {
+    console.log("values#", values?.length);
+  });
+  return {
+    [NAME]: str`Simple Value: ${
+      derive(values, (values) => values?.length || 0)
```

Comment:
`derive` calls `lift` underneath and for `lift` we already have support for schemas, so to build out your example, you might want to switch to that.

derive(inputs, fn) <=> lift(fn)(inputs)

so this could be `lift(inputschema, outputschema, (value) => value?.length || 0)(values)`